libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -0,0 +1,981 @@
+	return createASTNode<ModifierInvocation>(
+		_node,
+		createIdentifier(member(_node, "modifierName")),
+		member(_node, "arguments").isNull() ? nullptr : make_unique<std::vector<ASTPointer<Expression>>>(arguments)
+	);
+}
+
+ASTPointer<EventDefinition> ASTJsonImporter::createEventDefinition(Json::Value const&  _node)
+{
+	return createASTNode<EventDefinition>(
+		_node,
+		memberAsASTString(_node, "name"),
+		nullOrASTString(_node, "documentation"),
+		createParameterList(member(_node, "parameters")),
+		memberAsBool(_node, "anonymous")
+	);
+}
+
+ASTPointer<ElementaryTypeName> ASTJsonImporter::createElementaryTypeName(Json::Value const& _node)
+{
+	unsigned short firstNum;
+	unsigned short secondNum;
+
+	astAssert(_node["name"].isString(), "Expected 'name' to be a string!");
+
+	string name = member(_node, "name").asString();
+	Token token;
+	tie(token, firstNum, secondNum) = TokenTraits::fromIdentifierOrKeyword(name);
+	ElementaryTypeNameToken elem(token, firstNum,  secondNum);
+
+	std::optional<StateMutability> mutability = {};
+	if (_node.isMember("stateMutability"))
+		mutability = stateMutability(_node);
+
+	return createASTNode<ElementaryTypeName>(_node, elem, mutability);
+}
+
+ASTPointer<UserDefinedTypeName> ASTJsonImporter::createUserDefinedTypeName(Json::Value const& _node)
+{
+	astAssert(_node["name"].isString(), "Expected 'name' to be a string!");
+
+	vector<ASTString> namePath;
+	vector<string> strs;
+	string nameString = member(_node, "name").asString();
+	boost::algorithm::split(strs, nameString, boost::is_any_of("."));
+	for (string s: strs)
+		namePath.push_back(ASTString(s));
+	return createASTNode<UserDefinedTypeName>(
+		_node,
+		namePath
+	);
+}
+
+ASTPointer<FunctionTypeName> ASTJsonImporter::createFunctionTypeName(Json::Value const&  _node)
+{
+	return createASTNode<FunctionTypeName>(
+		_node,
+		createParameterList(member(_node, "parameterTypes")),
+		createParameterList(member(_node, "returnParameterTypes")),
+		visibility(_node),
+		stateMutability(_node)
+	);
+}
+
+ASTPointer<Mapping> ASTJsonImporter::createMapping(Json::Value const&  _node)
+{
+	return createASTNode<Mapping>(
+		_node,
+		createElementaryTypeName(member(_node, "keyType")),
+		convertJsonToASTNode<TypeName>(member(_node, "valueType"))
+	);
+}
+
+ASTPointer<ArrayTypeName> ASTJsonImporter::createArrayTypeName(Json::Value const&  _node)
+{
+	return createASTNode<ArrayTypeName>(
+		_node,
+		convertJsonToASTNode<TypeName>(member(_node, "baseType")),
+		nullOrCast<Expression>(member(_node, "length"))
+	);
+}
+
+ASTPointer<InlineAssembly> ASTJsonImporter::createInlineAssembly(Json::Value const& _node)
+{
+	astAssert(_node["evmVersion"].isString(), "Expected evmVersion to be a string!");
+	auto evmVersion = langutil::EVMVersion::fromString(_node["evmVersion"].asString());
+	astAssert(evmVersion.has_value(), "Invalid EVM version!");
+
+	yul::Dialect const& dialect = yul::EVMDialect::strictAssemblyForEVM(evmVersion.value());
```

Comment:
I think we should check that it matches the currently configured evm version. I don't think it works to switch the version just for single inline assembly blocks.